todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/ui/StatisticsViews.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016, The Android Open Source Project
+ * Copyright (c) 2017-2018 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.android.architecture.blueprints.todoapp.statistics.ui
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import com.example.android.architecture.blueprints.todoapp.R
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsEvent
+import com.example.android.architecture.blueprints.todoapp.statistics.StatisticsState
+import com.example.android.architecture.blueprints.todoapp.util.onAccept
+import com.spotify.mobius.Connectable
+import com.spotify.mobius.Connection
+import com.spotify.mobius.functions.Consumer
+
+class StatisticsViews(inflater: LayoutInflater, parent: ViewGroup) : Connectable<StatisticsState, StatisticsEvent> {
+
+    val rootView: View = inflater.inflate(R.layout.statistics_frag, parent, false)
+
+    private val statisticsTV: TextView
+
+    init {
+        statisticsTV = rootView.findViewById(R.id.statistics)
+    }
+
+    override fun connect(output: Consumer<StatisticsEvent>): Connection<StatisticsState> {
+        return onAccept<StatisticsState> { renderState(it) }.onDispose { }
+    }
+
+    private fun renderState(state: StatisticsState) {
+        when (state) {
```

Comment:
The compiler does not check if `when` is exhaustive unless it is used as an expression. I would say either return it or use `when {...}.let {}`

Reply:
@jeppes I see a warning in the IDE saying it should be exhaustive or have an else branch. What happens if you don't do either and run this?

Reply:
Did you get a warning in this case specifically? It works fine for me as long as I cover the the three cases (`Loading`, `Loaded`, and `Failed`).
Nothing would happen if you were to forget a case and run it, but "nothing happening" is typically not the behavior you want when switching on cases like this which can lead to some hard to find bugs.